framework/validators/DateValidator.php

```diff
@@ -265,6 +272,12 @@ public function init()
      */
     public function validateAttribute($model, $attribute)
     {
+        foreach (['min', 'max'] as $val) {
+            if ($this->$val instanceof \Closure) {
+                $this->$val = call_user_func($this->$val, $model);
+                $this->setMinMax($val);
+            }
+        }
```

Comment:
This doesn't look right. Only first validation will work correctly, second will just use `min` and `max` set by previous validation.

Reply:
didn't get it, example?

Reply:
```php
if (!$mode->validate()) {
    $model->minClosureDependency = 123;
    $model->validate();
}
```

`$minClosureDependency` change will be ignored by `$min` Closure in second validation, because it is no longer Closure - it was changed to fixed value in first validation. If Closure depends on `$minClosureDependency`, it will give you unpredictable results.